src/compas_fab/robots/robot.py

```diff
@@ -1313,25 +1314,22 @@ def forward_kinematics(self, configuration, group=None, use_attached_tool_frame=
 
         return frame_WCF
 
-    def plan_cartesian_motion(
-        self, frames_WCF, start_configuration=None, group=None, use_attached_tool_frame=True, options=None
-    ):
+    def plan_cartesian_motion(self, waypoints, start_configuration=None, group=None, options=None):
         """Calculate a cartesian motion path (linear in tool space).
 
         Parameters
         ----------
-        frames_WCF : :obj:`list` of :class:`compas.geometry.Frame`
-            The frames through which the path is defined.
+        waypoints : :class:`compas_fab.robots.Waypoints`
+            The waypoints for the robot to follow.
+            If a tool is attached to the robot, the :meth:`~compas_fab.robots.Waypoints.tool_coordinate_frame` parameter
+            should be set.
         start_configuration : :class:`compas_robots.Configuration`, optional
             The robot's full configuration, i.e. values for all configurable
-            joints of the entire robot, at the starting position. Defaults to
-            the all-zero configuration.
+            joints of the entire robot, at the start of the motion.
+            Defaults to the all-zero configuration.
         group : :obj:`str`, optional
-            The planning group used for calculation. Defaults to the robot's
-            main planning group.
-        use_attached_tool_frame : :obj:`bool`, optional
-            If ``True`` and there is a tool attached to the planning group, it will use its TCF
-            instead of the T0CF to calculate cartesian paths. Defaults to ``True``.
```

Comment:
I'm not convinced with the removal of this. I think we need to review the idea of having `tool_coordinate_frame` in the targets and waypoints classes. It feels to me like these could contain the name or key of the tool to use, but the coordinate frame should not be attached inside them, otherwise, if we serialize that, and re-measure the tool, the waypoints need to be somehow retrofitted/updated with the new tcf. That doesn't make much sense.